xchange-gdax/src/main/java/org/knowm/xchange/gdax/service/GDAXMarketDataServiceRaw.java

```diff
@@ -90,7 +97,74 @@ public GDAXProductBook getGDAXProductOrderBook(CurrencyPair currencyPair, int le
       throw handleError(e);
     }
   }
+  
+  static Instant lastCall = null;
+  static boolean rateLimited = false;
+  
+  public GDAXTrades getGDAXTradesExtended(CurrencyPair currencyPair, Long after, Integer limit) throws IOException {	
+	for (;;) {
+	    try {
+	      if (rateLimited) {
+	    	  long delta = Duration.between(lastCall, Instant.now()).toMillis();
+	    	  log.debug("Last call {} ms ago", delta);
+	    	  
+	    	  if (delta < 333) {
+	    		  try {
+	    			log.debug("Sleeping for {}ms", 333-delta);
+					Thread.sleep(333-delta);
+					
+				  } catch (InterruptedException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				  }
+	    	  } else if (delta > 3000) {
+	    		  log.debug("Clearing rate limiter");
+	    		  rateLimited = false;
+	    	  }
+	      }
+	    	
+	      lastCall = Instant.now();
+	      
+	      GDAXTrades gdaxTrades = this.gdax.getTradesPageable(
+	          currencyPair.base.getCurrencyCode(), 
+	          currencyPair.counter.getCurrencyCode(),
+	          after,
+	          limit);
+	      
+	      
+	      log.debug("GDAXTrades: earliest={}, latest={}, {}", 
+	    		  gdaxTrades.getEarliestTradeId(), 
+	    		  gdaxTrades.getLatestTradeId(), 
+	    		  gdaxTrades);
+	      
+	      return gdaxTrades;
+	    	
+	    } catch (GDAXException e) {
+	    	if (e.getHttpStatusCode() == 429) {
+		    	log.debug("Rate limit exceeded, sleeping for 1000ms");
+		    	
+		    	rateLimited = true;
+		    	
+		    	try {
+					Thread.sleep(1000);
+				} catch (InterruptedException e1) {
+					// TODO Auto-generated catch block
+					e1.printStackTrace();
+				}	    	
```

Comment:
was this PR build using maven before pushing?
seams that it was not because maven would fix the formatting

Reply:
Not, because the 'format' keyword doesn't appear to be recognised under my eclipse.  If there's a way to resolve that I'm happy to do so.

Reply:
ok

Reply:
You're quite right, it did.  Fixed.